### ISSUE TITLE
Analytics Implementation or QA Support for Accredited Representation Management - Appoint a Representative MVP

### ISSUE THREAD

**Issue by oddball-lindsay** [department-of-veterans-affairs/va.gov-team#96194]
# Google Analytics Implementation Request

### Please provide the following details for this request

- **Team Name**

    - Accredited Representation Management

- **Product Name**

    - Appoint a Representative
      
- **When is the launch date?**
    - Targeting December 2024

- **Provide the link(s) for Product/Page urls**
  
    - https://staging.va.gov/get-help-from-accredited-representative/appoint-rep/ is the start of the experience

- **What Events need to be tracked?** _NOTE: [Design System Components](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/Design-System-Components.1840808040.html) are automatically tracked and should be used whenever possible to avoid custom configuration_
    
    - We are hoping to track the OKRs [outlined in this mural, for Appoint a Representative MVP](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1718224425278/b300ea8d63dcaaa0cdad0ebf6b4a65a20fcc9371)

- **If applicable, what test users are needed?**  _NOTE: Do not put staging credentials in this ticket; store or reference them in a .md file in the `va.gov-team-sensitive` repository_

    - We have [test users related to our work outlined here](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/Administrative/vagov-users/staging-test-accounts-accredited-representation-management-poa.csv), if you'd like to see the authenticated experience

- **Any additional comments**

    - We are meeting tomorrow (10/31) to discuss metrics for all ARM overall, and I hope we can cover the Appoint a Rep MVP OKRs!

> Please leave the following blank

## Acceptance criteria
- [ ] All appropriate issue tagging is completed
- [x] QA has been completed
- [ ] Events have been added to `Product` and `Benefit Hub` Content Groups

## Additional information

Please refer to [Platform Collaboration Cycle](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/index.html) or the [Analytics Request Touchpoint](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Analytics-request.1782120453.html) on Platform website for more information about the Collaboration Cycle.

**Comment by oddball-lindsay** [comment]
Sharing our [Figma for Appoint a Rep](https://www.figma.com/design/bzbwObT9hiItve0q3cQX9c/Find-and-Appoint-a-Representative?node-id=455-76935&node-type=canvas&t=yIGr18vblBTYJKOF-0), to give context to the [third OKR](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1718224425278/b300ea8d63dcaaa0cdad0ebf6b4a65a20fcc9371?wid=0-1718224783508).

![Image](https://github.com/user-attachments/assets/95c453f0-a518-42ff-ac66-6cafab03aaa1)

**Comment by johnny-jesensky-adhoc** [comment]
Hi @oddball-lindsay -- We reviewed this in our refinement call today - are y'all using, or planning to use the Forms API to capture the responses? (So that you can compare those who have completed the form digitally to those that have been matched with a rep.

FYI @jestutt, @Blivaditis, @ChristineYoungZA.

**Comment by oddball-lindsay** [comment]
> So that you can compare those who have completed the form digitally to those that have been matched with a rep.

Sorry, not fully following -- @johnny-jesensky-adhoc which objective and key result are you referring to for this?

**Comment by oddball-lindsay** [comment]
@johnny-jesensky-adhoc I think the only time we'd be saving form responses is when an authenticated user has a form in progress, and this is part of the existing "Save in Progress" functionality. Once a form is completed and served up to the user as a PDF, I don't believe we store their responses beyond that.

When we move into v2 we likely will save the responses for any forms that are submitted digitally for the representatives to accept. But we won't be ready to chat about that until ~Feb 2025.

---

I do have an additional request for insight outside of the OKRs, would it be possible in Domo to track where users "drop off" from the form experience? By "drop off" I mean:
- Unauthenticated users leave the /appoint-rep experience overall (whether they go to a different page on VA.gov or leave VA.gov all together)
- Authenticated users let their form in progress lapse and do not finish completing it after the 60 day "Save in Progress" window.

While our main goal is form completion, it would be interesting to see if users leave the experience prior to form completion. If we see a large uptick in drop off on a particular screen, than this could indicate an opportunity to improve that particular step. 

Let me know if that scope can be included in this ticket, or if we should explore that tracking in a new ticket!

**Comment by oddball-lindsay** [comment]
It's been a little while since I've looked at this thread and wanted to leave some comments on where we're at now, especially since we've learned a lot from Datadog insights after our Appoint MVP release.

- We should be able to use Datatdog to get insights into how many users download a PDF, since we have 2 distinct endpoints for generating the 21-22 and 21-22a forms. 
- It seems like the Forms KPIs dashboard will be able to show insights to where users drop off, via the "Completion rate by step" module.

**Comment by jestutt** [comment]
@oddball-lindsay You're right that drop-off is available via the default completion rate module in the forms dashboard. 

@Basil Livaditis went thru and checked the tracking for each of the OKR tasks/tactics on this [Google sheet](https://docs.google.com/document/d/1mNjjLTuQ6KSL114J8rtol9Nale1W_IOOOD6zhqJeKjU/edit?usp=sharing). Note that we're going back in to review the checkboxes and will update the sheet shortly with that.

**Comment by Blivaditis** [comment]
Hi @oddball-lindsay, 
The privacy agreement check box event is being tracked, there are three other checkbox events that are not being tracked. One on step 2/4 (Your Information) "This address is on a United States military base outside of the U.S.", and the other two are on the Review Application page "I agree to the terms and conditions (*Required)", "I accept that this form will replace all my other VA Forms 21-22 and 21-22a (*Required)". If you would like these events tracked the component needs to be enabled.

<img width="805" alt="Image" src="https://github.com/user-attachments/assets/bd96a07b-f064-4d8b-be18-bfe39471fea4" />

**Comment by oddball-lindsay** [comment]
Thanks for the updates @jestutt and @B@Blivaditis 

Regarding the questions in the lower `The Appoint a Representative product remains "healthy"` section -- we've actually learned a lot about Datadog since this request was submitted and we are able to track those pieces in Datadog. Those items can be from this scope.

Regarding the checkboxes, it's good to know that those are not being tracked currently but that's also not a priority for us to monitor so I think it's fine to let that go from scope as well.

**Comment by jestutt** [comment]
@oddball-lindsay I'll close this ticket then.
